packages/components/src/DateField/DateField.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { HTMLAttributes, ReactElement, useRef } from 'react';
+import { useLocale } from '@react-aria/i18n';
+import { useDateFieldState } from '@react-stately/datepicker';
+import { AriaDateFieldProps, useDateField } from '@react-aria/datepicker';
+import { createCalendar, DateValue } from '@internationalized/date';
+import {
+  Box,
+  ThemeExtensionsWithParts,
+  useComponentStyles,
+  useStateProps,
+} from '@marigold/system';
+import { DateSegment } from './DateSegment';
+import { AriaButtonProps } from '@react-aria/button';
+import { FieldBase } from '../FieldBase';
+import { mergeProps } from '@react-aria/utils';
+
+// Props
+// ----------------
+export interface DateFieldThemeExtension
+  extends ThemeExtensionsWithParts<
+    'DateField',
+    ['segment', 'placeholder', 'field', 'calendarButton']
+  > {}
+
+export interface DateFieldProps
+  extends Omit<
+    AriaDateFieldProps<DateValue>,
+    'isDisabled' | 'isReadOnly' | 'isRequired'
+  > {
+  onChange?: (value: DateValue) => void;
+  value?: DateValue;
+  defaultValue?: DateValue;
+  ref?: React.RefObject<unknown> | undefined;
+  icon?: ReactElement;
+  action?: ReactElement;
+  buttonProps?: AriaButtonProps<'button'>;
+  isPressed?: boolean;
+  error?: boolean;
+  errorMessageProps?: HTMLAttributes<HTMLElement>;
+  disabled?: boolean;
+  readOnly?: boolean;
+  required?: boolean;
+  variant?: string;
+  size?: string;
+}
+
+// Component
+// ----------------
+
+export const DateField = ({
+  disabled,
+  readOnly,
+  required,
+  isPressed,
+  error,
+  errorMessage,
+  errorMessageProps,
+  buttonProps,
+  variant,
+  size,
+  icon,
+  action,
+  ...res
+}: DateFieldProps) => {
+  const { locale } = useLocale();
+  const props: AriaDateFieldProps<DateValue> = {
+    isDisabled: disabled,
+    isReadOnly: readOnly,
+    isRequired: required,
+    ...res,
+  };
+  const { label, description } = props;
+  const state = useDateFieldState({
+    ...props,
+    locale,
+    createCalendar,
+  });
+
+  const ref = useRef(null);
+  const { fieldProps, labelProps, descriptionProps } = useDateField(
+    props,
+    state,
+    ref
+  );
+  const styles = useComponentStyles(
+    'DateField',
+    {},
```

Comment:
here you should add variant and size. even if it's not used yet, but it's consistent.